Updated Organization param from location to Address Type.

Anubhav-2003 opened the pull request:
**What kind of change does this PR introduce?**

<!-- E.g. a bugfix, feature, refactoring, etc… -->
A feature

**Issue Number:**
Fixes:  #1650 

**Did you add tests for your changes?**
Yes
<!--Yes or No. Note: Add unit tests or automation tests for your code.-->

**Snapshots/Videos:**

<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**

<!--Add link to Talawa-Docs.-->

**Summary**

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**
Yes
<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->
The PR changes the location field in Organizations to AddressInput Type. All front-end applications using the API must change their queries, mutations, forms, components to accommodate for the change. 
**Other information**

<!--Add extra information about this PR here-->
The PR is still in work. Creating PR for review, and suggestions.
**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

<!--Yes or No-->
Yes

Anubhav-2003 commented:
@palisadoes Respected sir, I have created this PR for your review, and also update with my status. It is still under work. I have tried to make changes from location to address in Organizations in all the places I could figure out. Yet, I may be missing some places. Also, I have reduced the sample data for testing purposes,  adhering to the new structure of the organization type, but it is only temporary and I will put all the existing organizations with the new address field before merge. 

I have also updated and written some new test cases for the createOrganiztions.ts file, I will be writing more finally after getting your feedback on the actual file changes. Also I tried to make some changes in the Talawa-admin repo, but found out that there are a lot of files where this change is needed, before I can actually verify if the functionality is working properly. 

So someone will have to take the initiative of making all the changes in the Talawa-admin, which I think are plenty in number. Because unless the front-end sends the correct requests to the API, I am not sure about the changes. 

Thank you.

palisadoes commented:
Please fix the failing tests and let us know when you are ready for a review

Anubhav-2003 commented:
@palisadoes Sir, I have fixed the failing test cases. Can my code be please reviewed and given some insight as to if I need to make some more changes.

Thank you.

palisadoes commented:
Reviewers have been assigned
Please fix the conflicting files

xoldyckk commented on sample_data/organizations.json:
```diff
@@ -11,72 +11,16 @@
+      "postalCode": "110001",
+      "sortingCode": "ABC-123",
+      "state": "Delhi"
+    },
```
why were some smaple organizations deleted here?

Anubhav-2003 replied:
Actually Sir, I removed the sample organisations, and just kept one updating its located param to address just for testing purposes. I will bring back all the existing organisations before the PR merges.

xoldyckk commented on src/typeDefs/inputs.ts:
```diff
@@ -144,7 +144,7 @@ export const inputs = gql`
     name: String!
     description: String!
-    location: String
+    address: AddressInput
```
@palisadoes can organizations choose to give no address information?

palisadoes replied:
Make it mandatory

Anubhav-2003 replied:
@palisadoes ok sir. I will change it to mandatory.

Anubhav-2003 replied:
@palisadoes @xoldyckk Sir, I have  made the address field mandatory for creation of organizations. Thank You.

xoldyckk replied:
@palisadoes which of the address fields [here](https://github.com/PalisadoesFoundation/talawa-api/blob/46771dbdb3013f9a548a5b331cd2247e26032462/src/typeDefs/types.ts#L361) should be mandatory?

Anubhav-2003 commented:
@palisadoes Respected sir, I have fixed all the merge conflicts. Actually the conflict was due to this PR #1643 which was merged yesterday, which added a new field 'userRegistrationRequired' to organization type and removed 'isPublic' field. But my PR had these fields. So I had to change my sections of the code accordingly. 

Thank you.

palisadoes commented:
Reviewers. This should be complete. Please take a look
- @xoldyckk
- @JamaicanFriedChicken
- @sumitra19jha
- @DMills27

xoldyckk commented on src/typeDefs/inputs.ts:
```diff
@@ -138,7 +138,7 @@ export const inputs = gql`
     name: String!
     description: String!
-    location: String
+    address: AddressInput
```
@Anubhav-2003 make it non nullable in the input

palisadoes replied:
None to make it easier for the admin, but `Country Code` should come from a list of enums when updated

Anubhav-2003 replied:
@xoldyckk Sir, I have made it non-nullable in my last commit.

Anubhav-2003 replied:
@palisadoes  @xoldyckk Sir, I think in the front-end we should use an API to get the country code, and then use enums to verify it in the backend. There has to be major changes in the Talawa Admin, from updating queries, mutations, to using API's to send the data to the backend before this PR can be merged. Otherwise the Talawa-Admin portal would break.

palisadoes commented:
Please fix the conflicting files

Anubhav-2003 commented:
@palisadoes @xoldyckk Respected sirs, I have resolved all the conflicts as well as brought back all the sample organizations, updating their structure to have address param instead of location.

Anubhav-2003 commented:
@palisadoes Any update on my PR sir. Also shall I start working on talawa admin to administer the changes required to work with these changes in the API ?

palisadoes commented:
This really needs attention. Please review.

1. @JamaicanFriedChicken
1. @sumitra19jha
1. @DMills27
1. @xoldyckk

Anubhav-2003 commented on sample_data/organizations.json:
```diff
@@ -36,12 +45,20 @@
     "image": null,
-    "creatorId": "64378abd85008f171cf2990d",
-    "updatedAt": "2023-04-13T05:16:52.827Z",
+    "creator": "64378abd85008f171cf2990d",
```
@sumitra19jha Actually I did not make that change, while I was working in the issue some other PR got merged into develop branch, that had made the change. I just resolved the merge conflicts, keeping the changes in the updated develop. Even I was confused as to why it was made. My only change involves replacing location param with Address. 

Please let me know if I can help with anything else.

Thank You.

Anubhav-2003 commented:
@palisadoes Respected sir, Can I start working on the frontend implementation in Talawa-admin for these changes in the API. Unless those are made, this PR can not be merged. Otherwise the admin portal would break.

Thank You.

sumitra19jha commented:
Good to merge

Anubhav-2003 commented:
@palisadoes Respected sir, please don't merge it. Let me make the changes in the front end talawa admin first. Or else it will break. 

Thank you.

palisadoes commented:
When will you submit the Admin PR? Getting reviewers for resubmitting this PR once again is going to add delays

Anubhav-2003 commented:
> When will you submit the Admin PR? Getting reviewers for resubmitting this PR once again is going to add delays

Sir, as proposed by me earlier in the issue #1650, you approved for the issues to be solved simultaneously after this PR got approved before merging. But sir, there are multiple places in the frontend where changes are required which must be tackled by the issue https://github.com/PalisadoesFoundation/talawa-admin/issues/1398, like updating the UI, mutations, queries, and most importantly calling an API, to fetch the actual address data in the frontend to send the required fields. I can make those changes, but i am not assigned to this issue. If you authorise me i can make those changes. For now the API server work is completely done, but unless the frontend changes are made the API will throw an error. 

Thank you.

palisadoes commented:
OK. I'll let @Anvita0305 know to base the changes in that PR on this one

Anubhav-2003 commented:
@palisadoes ok sir, but if there are any delays on that issue, I will be more than obliged to make the changes myself in the admin portal, by creating an issue and asking for assignment. I will try to make the changes as early as possible. Please let me know if you need me to do that.

Thank you.

palisadoes commented:
@Anubhav-2003 you can ask to be assigned this issue
- https://github.com/PalisadoesFoundation/talawa-admin/issues/1457
